src/dragonfly_reporter/models.py

```diff
@@ -0,0 +1,29 @@
+from __future__ import annotations
+
+from enum import Enum
+from typing import Any, Optional
+
+from pydantic import BaseModel, model_validator
+
+
+class ObservationKind(Enum):
+    DependencyConfusion = "is_dependency_confusion"
+    Malware = "is_malware"
+    Spam = "is_spam"
+    Other = "something_else"
+
+
+class Observation(BaseModel):
+    kind: ObservationKind
+    summary: str
+    inspector_url: Optional[str]
+    extra: dict[str, Any] = {}
+
+    @model_validator(mode="after")
+    def model_validator(self: Observation) -> Observation:
+        if self.kind == ObservationKind.Malware:
+            assert (
```

Comment:
This is totally a nit, but why the assert here?

Reply:
It is required for Pydantic custom validators to work: https://docs.pydantic.dev/latest/concepts/validators/#field-validators

Reply:
gotcha